src/store/registration/saga.ts

```diff
@@ -169,6 +172,30 @@ export function* updateProfile(action) {
       yield put(setFirstTimeLogin(true));
       yield call(completeUserLogin);
       yield put(setStage(RegistrationStage.Done));
+
+      const inviterMatrixId = response.response.inviter.matrixId;
+      const inviterUserId = response.response.inviter.id;
+      const userData = yield call(getZEROUsersAPI, [inviterMatrixId]);
+
+      const normalizeUserData = (userData) => {
+        return {
+          userId: userData.id,
+          firstName: userData.profileSummary.firstName,
+          profileImage: userData.profileSummary.profileImage,
+          matrixId: userData.matrixId,
+        };
+      };
+
+      if (userData && userData.length > 0) {
+        yield put(receive({ users: { [inviterUserId]: normalizeUserData(userData[0]) } }));
+      }
+
+      if (inviterUserId) {
```

Comment:
Should this guard clause be higher? If it's empty then we just tried to fetch a user that doesn't exist and put it into the state, I think.

Reply:
Makes sense - Updated